SimFastTiming/FastTimingCommon/src/BTLBarDeviceSim.cc

```diff
@@ -93,28 +93,34 @@ void BTLBarDeviceSim::getHitsResponse(const std::vector<std::tuple<int,uint32_t,
     // --- Get the simHit time of arrival
     float toa = std::get<2>(hitRef);
 
-    // --- Accumulate the energy of simHits in the same crystal
-    if ( toa < bxTime_ ){  // this is to simulate the charge integration in a 25 ns window
-      (simHitIt->second).hit_info[0][0] += Npe;
-      (simHitIt->second).hit_info[0][1] += Npe;
+    if ( toa > bxTime_ || toa < 0 ) //just consider BX==0
+      continue;
+ 
+    // --- Accumulate the energy of simHits in the same crystal for the BX==0
+    // this is to simulate the charge integration in a 25 ns window
+    (simHitIt->second).hit_info[0][0] += Npe;
+    (simHitIt->second).hit_info[0][1] += Npe;
+ 
+    double distR = 0.5*topo.pitch().second - 0.1*hit.localPosition().y();
```

Comment:
these calculations should be done in an "else" statement after the "if" statement below (otherwise the calculations can be done twice for no reason)
this can be addressed in a followup PR

Reply:
Indeed. I did not change it to keep it as much as "before", but there is no need to do this twice (anyway our reference D35 will not enter in the if statement)